org.eclipse.mylyn.monitor.ui/src/org/eclipse/mylyn/monitor/ui/AbstractUserInteractionMonitor.java

```diff
@@ -95,70 +78,37 @@ protected void handleElementEdit(IWorkbenchPart part, Object selectedElement, bo
 	/**
 	 * Intended to be called back by subclasses.
 	 */
-	protected void handleNavigation(IWorkbenchPart part, Object targetElement, String kind, boolean contributeToContext) {
+	protected void handleNavigation(IWorkbenchPart part, Object targetElement, String kind,
+			boolean contributeToContext) {
 		handleNavigation(part.getSite().getId(), targetElement, kind, contributeToContext);
 	}
 
 	/**
 	 * Intended to be called back by subclasses. *
-	 * 
+	 *
 	 * @since 3.1
 	 */
 	protected void handleNavigation(String partId, Object targetElement, String kind, boolean contributeToContext) {
-		AbstractContextStructureBridge adapter = ContextCore.getStructureBridge(targetElement);
-		if (adapter.getContentType() != null) {
-			String handleIdentifier = adapter.getHandleIdentifier(targetElement);
-			InteractionEvent navigationEvent = new InteractionEvent(InteractionEvent.Kind.SELECTION,
-					adapter.getContentType(), handleIdentifier, partId, kind);
-			if (handleIdentifier != null && contributeToContext) {
-				ContextCore.getContextManager().processInteractionEvent(navigationEvent);
-			}
-			MonitorUiPlugin.getDefault().notifyInteractionObserved(navigationEvent);
-		}
+		throw new UnsupportedOperationException("should be handeled in AbstractContextInteractionMonitor");
 	}
 
 	/**
 	 * Intended to be called back by subclasses.
-	 * 
+	 *
 	 * @since 3.1
 	 */
 	protected void handleElementEdit(String partId, Object selectedElement, boolean contributeToContext) {
-		if (selectedElement == null) {
-			return;
-		}
-		AbstractContextStructureBridge bridge = ContextCore.getStructureBridge(selectedElement);
-		String handleIdentifier = bridge.getHandleIdentifier(selectedElement);
-		InteractionEvent editEvent = new InteractionEvent(InteractionEvent.Kind.EDIT, bridge.getContentType(),
-				handleIdentifier, partId);
-		if (handleIdentifier != null && contributeToContext) {
-			ContextCore.getContextManager().processInteractionEvent(editEvent);
-		}
-		MonitorUiPlugin.getDefault().notifyInteractionObserved(editEvent);
+		throw new UnsupportedOperationException("should be handeled in AbstractContextInteractionMonitor");
```

Comment:
This could be very surprising implementation change

Reply:
With next commit we only report this as a Warning in the Log.